go/kbfs/libkbfs/folder_branch_ops.go

```diff
@@ -4243,17 +4238,21 @@ const (
 	CtxAllowNameKey CtxAllowNameKeyType = iota
 )
 
-func checkDisallowedPrefixes(ctx context.Context, name string) error {
+func checkDisallowedPrefixes(
+	ctx context.Context, name data.PathPartString) error {
 	for _, prefix := range disallowedPrefixes {
-		if strings.HasPrefix(name, prefix) {
+		if strings.HasPrefix(name.Plaintext(), prefix) {
 			if allowedName := ctx.Value(CtxAllowNameKey); allowedName != nil {
 				// Allow specialized KBFS programs (like the kbgit remote
 				// helper) to bypass the disallowed prefix check.
-				if name == allowedName.(string) {
+				if name.Plaintext() == allowedName.(string) {
 					return nil
 				}
 			}
-			return DisallowedPrefixError{name, prefix}
+			// Put the plaintext in the error if it starts with a
+			// disallowed prefix -- there's no expectation of privacy
+			// in that case.
+			return DisallowedPrefixError{name.Plaintext(), prefix}
```

Comment:
Good question.  Hmm.  It would, and this was probably written before I thought about the `.kbfs_fileinfo_` restriction.  But, this method is only used on file create/rename events, where we disallow `kbfs_` completely.

That said, I don't have a good reason for doing that here, so I'll change that error type to contain a PPS.